canceled/canceled.go

```diff
@@ -0,0 +1,26 @@
+package grpccanceled
+
+import (
+	"context"
+	"strings"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+// UnaryServerInterceptor implements the UnaryServerInterceptor interface.
+// It maps grpc codes from one code to another.
+func UnaryServerInterceptor() grpc.UnaryServerInterceptor {
+	return func(ctx context.Context, req interface{}, _ *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
+		resp, err := handler(ctx, req)
+		if err != nil {
+			// check end of string matches context cancelled string (if so - make it codes.Cancelled)
+			if strings.HasSuffix(err.Error(), "context canceled") {
+				return nil, status.Error(codes.Canceled, err.Error())
+			}
+			return nil, err
+		}
+		return resp, nil
```

Comment:
I am wondering whether it would be wise to return the resp and the error in either case. I think you are doing more than you say you do by not doing that. I also think we can do a better match than just that.

Reply:
We have the check above for if `err != nil`, so we know that err is nil at this point. However I think you're right, we can add resp to the above returns (though I'm not 100% it will make a difference.